python/ray/data/dataset.py

```diff
@@ -2213,12 +2213,18 @@ def make_generator():
 
         return TorchIterableDataset(make_generator)
 
+    FeatureTypeSpec = Union[
```

Comment:
Wasn't sure where to place this. Needed to introduce a new type because otherwise the type annotation for `output_signature` gets gnarly.